### ISSUE TITLE
Possible heavy recursion?

### ISSUE THREAD

**Issue by schuemie** [OHDSI/SqlRender#352]
There have been reports of errors that seem to be related to running out of stack memory: https://github.com/OHDSI/CohortDiagnostics/issues/1051

If you really need 3MB to handle a simple query there might be an issue with too many recursions.

**Comment by schuemie** [comment]
[This](https://community.rstudio.com/t/error-in-jcheck-java-exception-no-description-because-tostring-failed/161960/5) suggests the problem may be the JDBC driver (since they're not using SqlRender, but still have the same problem)

**Comment by azimov** [comment]
I don't think this is a SqlRender issue. Using SqlRender with the DBI drivers instead of the JDBC driver fixed this issue for me, though this is not a workable solution in many situations (e.g. with RSQLite on its own using "main" schema fails and requires `dbplyr::in_schema` calls).

**Comment by schuemie** [comment]
Adding `options(java.parameters = "-Xss3m")`  to global.R (or .Rprofile) seems to be a viable workaround

**Comment by schuemie** [comment]
Doesn't seem to be related to SqlRender
